File: apps/otel-ts/src/routes/root.ts
// SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
// SPDX-License-Identifier: Apache-2.0

import { FastifyInstance, FastifyRequest, FastifyReply } from "fastify";
import { getLogger } from "../otel/logging";

export async function registerRootRoutes(
  fastify: FastifyInstance,
): Promise<void> {
  fastify.get("/", async (request: FastifyRequest, reply: FastifyReply) => {
    const logger = getLogger();
    logger.emit({
      severityText: "info",
      body: "Root handler",
    });
    return reply.send("OpenTelemetry Lab / Typescript");
  });
}

File: apps/otel-rust/src/routes/root.rs
// SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
// SPDX-License-Identifier: Apache-2.0

use opentelemetry::global;
use opentelemetry::trace::{Span, Tracer};
use opentelemetry::KeyValue;
use tracing::info;

pub async fn handler_root() -> &'static str {
    info!(target: "otel_rust::handlers", "Root handler");
    let tracer = global::tracer("otel-rust");
    let mut span = tracer.start("handle request");
    span.set_attribute(KeyValue::new("http.route", "/".to_string()));
    span.end();
    "OpenTelemetry Lab / Rust"
}